src/nimibook/types.nim

```diff
@@ -11,11 +11,33 @@ type
     isActive*: bool
   Toc* = object
     title*: string
-    path*: string    
+    path*: string
     entries*: seq[Entry]
 
-proc url*(e: Entry): string = 
+proc nimOutput*(entry: Entry, rootfolder: string) =
+  let
+    cmd = "nim"
+    args = ["r", "-d:release", "-d:nimibCustomPostInit", joinPath(rootfolder, entry.path)]
+  # debugEcho "[Executing] ", cmd, " ", args.join(" ")
+  if execShellCmd(cmd & " " & args.join(" ")) != 0:
+    quit(1)
+
+proc mdOutput*(entry: Entry, rootfolder: string) =
+  raise newException(IOError, "Markdown not yet supported. We advise listening to elevators music while we are working on this feature.")
+
+proc output*(entry: Entry, rootfolder: string) =
```

Comment:
what about naming this `publish` instead of `output` (and consequently have nimPublish, mdPublish)? I love nim overloading :)
also with respect to signature, maybe it make sense to incorporate rootfolder in entry path early on during the dsl so that we do not need to keep manage it here.